File: 941. Valid Mountain Array.py
class Solution:
    def validMountainArray(self, arr: List[int]) -> bool:
        up = False
        dn = False
        prev = arr[0]
        for i in range(1, len(arr)):
            if prev < arr[i]:
                up = True
                if dn:
                    return False
            elif prev > arr[i]:
                dn = True
                if not up:
                    return False
            else:
                return False
            prev = arr[i]
        if up and dn:
            return True
        else:
            return False

File: 941. Valid Mountain Array.go
func validMountainArray(arr []int) bool {
    up := false
	dn := false
	prev := arr[0]
	for i := 1; i < len(arr); i++ {
		if prev < arr[i] {
			up = true
			if dn == true {
				return false
			}
		} else if prev > arr[i] {
			dn = true
			if !up {
				return false
			}
		} else {
			return false
		}
		prev = arr[i]
	}

	if up && dn {
		return true
	} else {
		return false
	}
}